extra/modules/greenbids-real-time-data/src/main/java/org/prebid/server/hooks/modules/greenbids/real/time/data/config/GreenbidsRealTimeDataConfiguration.java

```diff
@@ -0,0 +1,153 @@
+package org.prebid.server.hooks.modules.greenbids.real.time.data.config;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import com.maxmind.geoip2.DatabaseReader;
+import io.vertx.core.Vertx;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.model.filter.ThrottlingThresholds;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.core.ThrottlingThresholdsFactory;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.core.GreenbidsInferenceDataService;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.core.FilterService;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.core.ModelCache;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.core.OnnxModelRunner;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.core.OnnxModelRunnerFactory;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.core.OnnxModelRunnerWithThresholds;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.core.ThresholdCache;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.core.GreenbidsInvocationService;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.v1.GreenbidsRealTimeDataProcessedAuctionRequestHook;
+import org.prebid.server.json.ObjectMapperProvider;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+@ConditionalOnProperty(prefix = "hooks." + GreenbidsRealTimeDataModule.CODE, name = "enabled", havingValue = "true")
+@Configuration
+@EnableConfigurationProperties(GreenbidsRealTimeDataProperties.class)
+public class GreenbidsRealTimeDataConfiguration {
+
+    @Bean
+    DatabaseReader dbReader(GreenbidsRealTimeDataProperties properties) throws IOException {
+        URL url = new URL(properties.getGeoLiteCountryPath());
+        Path databasePath = Files.createTempFile("GeoLite2-Country", ".mmdb");
+
+        try (InputStream inputStream = url.openStream();
+             FileOutputStream outputStream = new FileOutputStream(databasePath.toFile())) {
+            inputStream.transferTo(outputStream);
+        }
+
+        try {
+            return new DatabaseReader.Builder(databasePath.toFile()).build();
+        } catch (IOException e) {
+            throw new PreBidException("Failed build DatabaseReader from DB", e);
+        }
```

Comment:
Maybe it's conceptually better to create a proxy (or factory) for `DatabaseReader` that also implements the `Initializable` interface? The application will still wait for the `initialize` method to complete before it can start working, but this will be easier to extend if additional logic is needed to download this file.

Besides, I just don't want to download anything while we're constructing beans :p

@And1sS what is your opinion on it?

Reply:
Added `DatabaseReaderFactory` that implements Initializable with `AtomicReference<DatabaseReader> databaseReaderRef` to hold the DatabaseReader. 

Provided into `GreenbidsInferenceDataService` the factory and then get the DatabaseReader at `getCountry` from atomic reference